src/service/impl/departures/utils/favorites.ts

```diff
@@ -0,0 +1,76 @@
+import { EstimatedCall } from '../../../../graphql/journeyplanner-types_v3';
+import { FavoriteDeparture } from '../../../types';
+import { QuayDeparturesQuery } from '../gql/jp3/quay-departures.graphql-gen';
+import { StopPlaceQuayDeparturesQuery } from '../gql/jp3/stop-departures.graphql-gen';
+
+export function filterStopPlaceFavorites(
```

Comment:
I feel like functions like these with a lot of conditionals and checks are good cases for some sanity unit tests. Can also make it easier to understand intentions at a later point.